pyomo/network/port.py

```diff
@@ -0,0 +1,512 @@
+            # already exists before making a new one.
+            evar = eblock.component(name)
+            if evar is None:
+                evar = Var(index_set)
+                eblock.add_component(name, evar)
+            out_vars.append(evar)
+
+            # Create and potentially initialize split fraction variables.
+            # This function will be called for every Extensive member of this
+            # port, but we only need one splitfrac variable per arc, so check
+            # if it already exists before making a new one.
+            if eblock.component("splitfrac") is None:
+                eblock.splitfrac = Var()
+                splitfracspec = port.get_split_fraction(arc)
+                if splitfracspec is not None:
+                    eblock.splitfrac = splitfracspec[0]
+                    if splitfracspec[1]:
+                        eblock.splitfrac.fix()
+
+            # Create constraint for this member using splitfrac.
+            cname = "%s_split" % name
+            def rule(m, *args):
+                if len(args):
+                    return evar[args] == eblock.splitfrac * var[args]
+                else:
+                    return evar == eblock.splitfrac * var
+            con = Constraint(index_set, rule=rule)
+            eblock.add_component(cname, con)
+
+        if len(out_vars):
+            if write_var_sum:
+                # Create var total sum constraint: var == sum of evars
+                # Need to alphanum port name in case it is indexed.
+                cname = unique_component_name(port_parent, "%s_%s_outsum" %
+                    (alphanum_label_from_name(port.local_name), name))
+                def rule(m, *args):
+                    if len(args):
+                        return sum(evar[args] for evar in out_vars) == var[args]
+                    else:
+                        return sum(evar for evar in out_vars) == var
+                con = Constraint(index_set, rule=rule)
+                port_parent.add_component(cname, con)
+            else:
+                # OR create constraint on splitfrac vars: sum == 1
+                cname = unique_component_name(port_parent,
+                    "%s_frac_sum" % alphanum_label_from_name(port.local_name))
+                con = Constraint(expr=
+                    sum(a.expanded_block.splitfrac for a in port.dests()) == 1)
+                port_parent.add_component(cname, con)
+
+        return out_vars
+
+    # Python 2 compatibility
+    Equality = staticmethod(Equality)
+    Extensive = staticmethod(Extensive)
+    Combine = staticmethod(Combine)
+    Split = staticmethod(Split)
+
+
+class SimplePort(Port, _PortData):
+
+    def __init__(self, *args, **kwd):
+        _PortData.__init__(self, component=self)
+        Port.__init__(self, *args, **kwd)
+
+
+class IndexedPort(Port):
+    pass
+
+
+register_component(
+    Port, "A bundle of variables that can be connected to other ports.")
+
+
+class PortExpander(Plugin):
```

Comment:
Wait, what? What's the difference between `ArcExpander` and `PortExpander`? Also, same argument as before.

Reply:
Left this here by mistake, forgot to take it out. There is no port expander.